crates/net/kademlia.rs

```diff
@@ -155,6 +155,9 @@ impl KademliaTable {
         peer.is_proven = true;
         peer.last_pong = time_now_unix();
         peer.last_ping_hash = None;
+        if peer.revalidation.is_some() {
+            peer.revalidation = Some(true);
+        }
```

Comment:
Not sure if I like it, but maybe:
```suggestion
        peer.revalidation = peer.revalidation.and(Some(true));
        }
```

Reply:
Anyway, not sure how can we get into this case:
- peer.revalidation is None, and
- pong has been answered
🤔

Reply:
In case we have pinged a peer for the first time. We might also fall, though this is less common because of the revalidations, in case they ping us and we ping them back because their endpoint proof was expired. see [here](https://github.com/lambdaclass/ethereum_rust/blob/main/crates/net/net.rs#L128-L131).